src/library/basetracktablemodel.h

```diff
@@ -300,7 +300,8 @@ class BaseTrackTableModel : public QAbstractTableModel, public TrackModel {
 
     static int s_bpmColumnPrecision;
     static bool s_keyColorsEnabled;
-    static ColorPalette s_keyColorPalette;
+    // This need to be a pointer to avoid static initialization order issues
+    static std::unique_ptr<ColorPalette> s_pKeyColorPalette;
```

Comment:
I don't think the problem is static initialization. The only way that can occur if another global static is initialized with `s_pKeyColorPalette` in another translation unit before `s_pKeyColorPalette` is initialized. That can't happen because `s_pKeyColorPalette` is only accessible by `basetracktablemodel.cpp`. The only other way to access it is `BaseTrackTableModel::setKeyColorPalette` and `BaseTrackTableModel::roleValue` both of which do not get called before `main()` and thus `s_pKeyColorPalette` should be initialized at that point. 

I think there is something going on which we don't understand. Can you share the crashing stacktrace? 

Also, even if it was accessed before initialization, changing into to a pointer does not actually fix the associated undefined behavior. The object is still accessed before its initialization, the underlying memory is likely just the same bitpattern as `nullptr` so it doesn't crash in practice, but in theory the compiler could still optimize this into a crash because this relies on UB.